exporter/general_collector.go

```diff
@@ -54,16 +54,28 @@ func (d *generalCollector) collect(ch chan<- prometheus.Metric) {
 
 func mongodbUpMetric(ctx context.Context, client *mongo.Client, log *logrus.Entry) prometheus.Metric { //nolint:ireturn
 	var value float64
+	var clusterRole mongoDBNodeType
 
 	if client != nil {
 		if err := client.Ping(ctx, readpref.PrimaryPreferred()); err == nil {
 			value = 1
 		} else {
-			log.Errorf("error while checking mongodb connection: %s. mongo_up is set to 0", err)
+			log.Errorf("error while checking mongodb connection: %s. mongo_up is set to 0", err.Error())
+		}
+
+		if nodeType, err := getNodeType(ctx, client); err != nil {
```

Comment:
should we do another request to DB? I think we can pass this data from exporter.go during creation of general collector